src/framework/piral-core/src/utils/state.test.ts

```diff
@@ -0,0 +1,19 @@
+import * as hooks from '../hooks';
+import { withAll, withRootExtension } from './state';
+import { RootListener } from '../RootListener';
+import { createGlobalState } from '../state';
+
+describe('State Module', () => {
+  it('withRootExtension should create an extension', () => {
+    const dispatch = withRootExtension('testNmae', RootListener);
+    const result = dispatch({ registry: { extensions: {} } });
+    expect(result).not.toBe('undefined');
+    expect(result['registry']['extensions']['testNmae']).toBeTruthy();
+  });
+
+  it('withAll', () => {
+    const dispatcher = withAll((state) => state);
+    const result = dispatcher({ state: 'foo' });
+    expect(result).toEqual({ state: 'foo' });
```

Comment:
Not useful - use something more meaningful than the identity function. Also use multiple dispatchers (which is why `withAll` was brought in the first place, i.e., if you only do `withAll(foo)` then you can already just do `foo`; instead `withAll` makes sense with, e.g., `withAll(foo, bar, ...)`).